vec3.h

```diff
@@ -0,0 +1,120 @@
+//code by Ameya Vikrama Singh
+#include<iostream>
+#include<cmath>
+class vec3{
+    public:
+        float x,y,z;
+
+        vec3(){
+            x=y=z=0;
```

Comment:
This is bad programming practice. There is no reason to write it like this. It becomes simpler to read and, more importantly, reason about, if you write each one on its own line.

However, the modern C++ way of writing a constructor like this is using [member initializer lists](https://www.learncpp.com/cpp-tutorial/constructor-member-initializer-lists/).